chunk.go

```diff
@@ -0,0 +1,38 @@
+package functools
+
+import (
+	"math"
+)
+
+// Chunk consumes a slice of a generic type and
+// returns a slice composed of multiple chunks of a user-specified size
+//
+// Vacuously, empty slices return empty regardless of the chunk size.
+//
+// When the chunk size is zero or negative, return an empty slice
+func Chunk[T any](slice []T, size int) [][]T {
+	if size <= 0 {
+		return [][]T{}
+	}
+	n := len(slice)
```

Comment:
An edge case is when `n == 0`.  In the current logic, you return `{{}}` if `size > 0` and `{}` if `size <= 0`. I think it would make sense to return `{}` regardless of size if `n == 0` so I would modify this if-condition to `if size <= 0 || n == 0`. Please add a test case for this empty slice case too.

Reply:
Happy to discuss this if you think the logic should be otherwise, btw!